crates/biome_css_analyze/src/lint/nursery/no_unknown_unit.rs

```diff
@@ -0,0 +1,188 @@
+                                    .name()
+                                    .ok()?
+                                    .value_token()
+                                    .ok()?
+                                    .text_trimmed()
+                                    .to_lowercase();
+
+                                if strip_vendor_prefix(function_name.as_str()) == "image-set" {
+                                    allow_x = true;
+                                    break;
+                                }
+                            }
+                            CssSyntaxKind::CSS_GENERIC_PROPERTY => {
+                                let property_name = ancestor
+                                    .cast::<CssGenericProperty>()?
+                                    .name()
+                                    .ok()?
+                                    .as_css_identifier()?
+                                    .value_token()
+                                    .ok()?
+                                    .text_trimmed()
+                                    .to_lowercase();
+
+                                if property_name == "image-resolution" {
+                                    allow_x = true;
+                                    break;
+                                }
+                            }
+                            CssSyntaxKind::CSS_QUERY_FEATURE_PLAIN => {
+                                let feature_name = ancestor
+                                    .cast::<CssQueryFeaturePlain>()?
+                                    .name()
+                                    .ok()?
+                                    .value_token()
+                                    .ok()?
+                                    .text_trimmed()
+                                    .to_lowercase();
+
+                                if RESOLUTION_MEDIA_FEATURE_NAMES.contains(&feature_name.as_str()) {
+                                    allow_x = true;
+                                    break;
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+
+                    if !allow_x {
+                        return Some(NoUnknownUnitState {
+                            unit,
+                            span: unit_token.text_trimmed_range(),
+                        });
+                    }
+                }
+
+                None
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.span,
+                markup! {
+                    "Unexpected unknown unit: "<Emphasis>{ state.unit }</Emphasis>
+                },
+            )
+            .note(markup! {
+                "Fix to a known unit."
```

Comment:
It would be nice if we could list the **known** units

Reply:
If we decide to list the known units, what would be the best way to display them?

Here are some ideas I've considered:
1. Display a selection of the most commonly used known units.
2. List all known units, which could result in a large amount of data being displayed.
3. Show known units similar to the target unknown unit.

Please note that I am not a master of CSS units, so implementing ideas other than the first may be difficult for me.

Reply:
I'm sure in the future we'll need all units somewhere, but for now I think we can keep a short list of possible alternatives: px, em, rem, etc.

Let's make sure to use `etc.` so we tell the user there are more units. Ideally, we could also add a link that lists all units.

You can use this function to create a list: https://github.com/biomejs/biome/blob/main/crates%2Fbiome_analyze%2Fsrc%2Frule.rs#L751-L754

And you can use `<Hyperlink>` inside the markup to add a link.

Reply:
I have modified the `diagnostic` using `footer_list`.